src/input.rs

```diff
@@ -159,7 +159,13 @@ impl Action {
             Action::Paste |
             Action::PasteSelection => {
                 Clipboard::new()
-                    .and_then(|clipboard| clipboard.load_selection())
+                    .and_then(|clipboard|
+                              match *self {
```

Comment:
I had started down that route but wasn't having much luck, Rust n00b here, but I'll see if I can make those changes.

Thanks for the feedback!